src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1366,6 +1415,14 @@ private void onDecodeConnect(
                 keepAlive = packet.keepAlive();
                 keepAliveTimeout = Math.round(keepAlive * 1.5 * 1000);
                 doSignalKeepAliveTimeoutIfNecessary();
+                if (sessionsAvailable)
```

Comment:
Changing this to be based on `sessionExpiryInterval` has the added benefit of not storing state for sessions that have an expiry of zero, even when we have full support for session state.